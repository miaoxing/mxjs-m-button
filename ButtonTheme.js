import {View} from '@fower/taro';
import clsx from 'clsx';
import PropTypes from 'prop-types';
import './theme.scss';

// @experimental 可能重命名，移到其他包
const ButtonTheme = ({variant = 'pill', className, children, ...props}) => {
  // TODO 读取主题，显示不同的按钮主题
  return (
    <View className={clsx('mx-btn-theme', variant && ('mx-btn-theme-' + variant), className)} {...props}>
      {children}
    </View>
  );
};

ButtonTheme.propTypes = {
  variant: PropTypes.string,
  className: PropTypes.string,
  children: PropTypes.node,
};

export default ButtonTheme;
