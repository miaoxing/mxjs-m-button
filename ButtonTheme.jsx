import {View} from '@fower/taro';
import clsx from 'clsx';
import PropTypes from 'prop-types';
import './theme.scss';
import {useTheme} from '@mxjs/theme';

// @experimental 可能重命名，移到其他包
const ButtonTheme = ({variant, className, children, ...props}) => {
  const theme = useTheme();

  return (
    <View
      className={clsx(
        'mx-btn-theme',
        'mx-btn-theme-' + (variant || theme?.components?.ButtonTheme?.variant || 'pill'),
        className,
      )}
      {...props}
    >
      {children}
    </View>
  );
};

ButtonTheme.propTypes = {
  variant: PropTypes.string,
  className: PropTypes.string,
  children: PropTypes.node,
};

export default ButtonTheme;
