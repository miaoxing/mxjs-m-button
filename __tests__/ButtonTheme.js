import Button from '../';
import ButtonTheme from '../ButtonTheme';
import {render} from '@testing-library/react';

describe('ButtonTheme', () => {
  test('basic', () => {
    const {container} = render(<ButtonTheme>
      <Button>foo</Button>
    </ButtonTheme>);
    expect(container).toMatchSnapshot();
  });
});
