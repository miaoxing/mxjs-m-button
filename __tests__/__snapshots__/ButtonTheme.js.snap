// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonTheme basic 1`] = `
<div>
  <taro-view-core
    class="mx-btn-theme mx-btn-theme-pill"
  >
    <taro-button-core
      class="mx-btn"
    >
      foo
    </taro-button-core>
  </taro-view-core>
</div>
`;
